src/mapping/query/function/split.rs

```diff
@@ -23,6 +23,7 @@ impl SplitFn {
 }
 
 impl Function for SplitFn {
+    #[allow(clippy::collapsible_match)] // I expect this file to be going away shortly
```

Comment:
@JeanMertz @FungusHumungus is this accurate? It seems like `src/mapping` is largely vestigial at this point. Do you think we'd drop it as part of https://github.com/timberio/vector/pull/6353 ?

Reply:
Yeah, feel free to ignore this. I'll drop it once #6353 makes it into master (later today, as the CI is expected to be green within the hour).